### DISCUSSION TITLE
Strange issues when tif has locked layer

### DISCUSSION THREAD

**Discussion by paulc-us** [ImageMagick/ImageMagick#6535] [Digital Image Processing]
I have an ImageMagick script that works well to convert and compress an image to our specifications. I recently came across a tif that had strange results, where after converting, the macOS thumb of the image shows the image correctly, but when opened in preview or photoshop, only the text layer shows. 

If I open the original image, it has one background layer, and it is locked.  If I unlock and resave, the imagemagick script works as expected. How should this be handled?  Can imagemagick unlock a layer before it converts?

Here is the script:

```
#!/bin/bash

PROFILE="/Library/Application Support/Adobe/Color/Profiles/Recommended/sRGB Color Space Profile.icm"
convert="/usr/local/bin/convert"  # Use 'convert' instead of 'magick'

for image in "$@"
do
  FULLIMAGEPATH="$image"
  IMAGEPATH=$(dirname "$image")
  IMAGENAME=$(basename "$image" | sed 's/\.[^.]*$//')

  mkdir -p "${IMAGEPATH}/mediavalet"
  "$convert" \
    "$FULLIMAGEPATH" \
    -flatten \
    -resize "3000x3000>" \
    -density 100 \
    -compress lzw \
    -intent Relative \
    -black-point-compensation \
    -profile "$PROFILE" \
    -colorspace sRGB \
    ${IMAGEPATH}/mediavalet/${IMAGENAME}.tif 2>/dev/null
done
```

**Comment by fmw42** [comment]
Post the input file and your resulting output

**Comment by paulc-us** [comment]
Unfortunately, I cannot share the images.  I appreciate if you could help, but I understand if you can't, without the images. 

Does this sound familiar at all? Would a locked background layer cause an issue converting?

**Reply by fmw42** [reply]
Can you find or create an image that you can share that shows your problem?

**Comment by paulc-us** [comment]
In trying to replicate the issue on a new image, I see that this image has 2 alpha channels. If i run it with -alpha off, it works o this image. Are there any negative effects from keeping "alpha off" in the script for all images?

**Reply by fmw42** [reply]
It depends whether there are any of your images with transparency that you want to see.

**Comment by snibgo** [comment]
> Are there any negative effects from keeping "alpha off" in the script for all images?

`-alpha off` and `-flatten` do different things. If any pixels are fully transparent, `-alpha off` will use the colours of those pixels, fully opaque, but `-flatten` will replace the pixel's colour with the current background colour. Similarly for partly transparent pixels.
